packages/api/internal/handlers/sandboxes_list.go

```diff
@@ -316,11 +320,24 @@ func (a *APIStore) getSandboxesMetrics(
 	)
 
 	// Log operation summary
+	if len(metricsErrors) > 0 {
+		errorStrings := make([]string, len(metricsErrors))
+		for i, err := range metricsErrors {
+			errorStrings[i] = err.Error()
+		}
+
+		err := errors.Join(metricsErrors...)
+		a.logger.Error("Received errors while fetching metrics for some sandboxes",
+			zap.Int32("failed_fetches", errorCount.Load()),
+			zap.Error(err),
+			zap.Strings("individual_errors", errorStrings),
+		)
+	}
+
 	a.logger.Info("Completed fetching sandbox metrics",
```

Comment:
this should be in else case, even better if you can do the positive case `len()==0` as first